File: backend/crawl/app/models/pitcher_stats.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.models.database import Base


class PitcherStats(Base):
    __tablename__ = "pitcher_stats"

    id = Column(Integer, primary_key=True, index=True)
    year = Column(String(10))
    player_id = Column(Integer, ForeignKey("player.id"))
    team_id = Column(Integer, ForeignKey("team.id"))
    era = Column(String(10))
    game_count = Column(Integer)
    win = Column(Integer)
    lose = Column(Integer)
    save = Column(Integer)
    hld = Column(Integer)
    wpct = Column(String(10))
    ip = Column(String(10))
    h = Column(Integer)
    hr = Column(Integer)
    bb = Column(Integer)
    hbp = Column(Integer)
    so = Column(Integer)
    r = Column(Integer)
    er = Column(Integer)
    whip = Column(String(10))
    
    player = relationship("Player", back_populates="pitcher_stats")
    team = relationship("Team", back_populates="pitcher_stats")

File: backend/baseball/src/main/java/com/newbie/baseball/domain/playerStats/entity/PitcherStats.java
package com.newbie.baseball.domain.playerStats.entity;

import com.newbie.baseball.domain.player.entity.Player;
import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "pitcher_stats")
public class PitcherStats {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "year", length = 10)
    private String year;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "player_id")
    private Player player;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Column(name = "era", length = 10)
    private String era;

    @Column(name = "game_count")
    private Integer gameCount;

    @Column(name = "win")
    private Integer win;

    @Column(name = "lose")
    private Integer lose;

    @Column(name = "save")
    private Integer save;

    @Column(name = "hld")
    private Integer hld;

    @Column(name = "wpct", length = 10)
    private String wpct;

    @Column(name = "ip", length = 10)
    private String ip;

    @Column(name = "h")
    private Integer h;

    @Column(name = "hr")
    private Integer hr;

    @Column(name = "bb")
    private Integer bb;

    @Column(name = "hbp")
    private Integer hbp;

    @Column(name = "so")
    private Integer so;

    @Column(name = "r")
    private Integer r;

    @Column(name = "er")
    private Integer er;

    @Column(name = "whip", length = 10)
    private String whip;
}
